log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+    } else if (config.color != -1 && config.mode == "group") {
+        marks.push(getGroupBarMark(config, this.metadata));
+    } else {
+        marks.push(getBarMark(config, this.metadata));
+    }
+
+    if (config.range) {
+        signals = getRangeSignals(config, signals);
+        marks = getRangeMark(config, marks);
+    }
+
+    this.spec.width = config.width;
+    this.spec.height = config.height;
+    this.spec.axes = axes;
+    this.spec.data = dataTable;
+    this.spec.scales = scales;
+    this.spec.padding = config.padding;
+    this.spec.marks = marks;
+    this.spec.signals = signals;
+
+    var specc = JSON.stringify(this.spec);
+};
+
+bar.prototype.draw = function(div, callbacks) {
+    if(this.config.maxLength != -1){
+        var dataset = this.spec.data[0].values;
+        var maxValue = this.config.maxLength;
+        if(dataset.length >= this.config.maxLength){
+            var allowedDataSet = [];
+            var startingPoint = dataset.length - maxValue;
+            for(var i = startingPoint; i < dataset.length;i++){
+                allowedDataSet.push(dataset[i]);
+            }
+            this.spec.data[0].values = allowedDataSet;
+        }
+    }
+
+    drawChart(div, this, callbacks);
+};
+
+bar.prototype.insert = function(data) {
+
+    var xAxis = this.metadata.names[this.config.x];
+    var yAxis = this.metadata.names[this.config.y];
+    var size = this.metadata.names[this.config.size];
+    var color = this.metadata.names[this.config.color];
+
+    if (this.config.maxLength != -1 && this.config.maxLength <  (this.view.data(this.config.title).values().length + data.length)) {
+
+        var allDataSet = this.view.data(this.config.title).values().concat(data);
+        var allowedRemovableDataSet = [];
+        for (i = 0; i < allDataSet.length - this.config.maxLength; i++) {
+            allowedRemovableDataSet.push(this.view.data(this.config.title).values()[i][xAxis]);
+        }
+
+        for (i = 0; i < data.length; i++) {
+            var isValueMatched = false;
+            this.view.data(this.config.title).update(function(d) {
+                    if (color == null) {
+                        return d[xAxis] == data[i][xAxis];
+                    } else {
+                        return d[xAxis] == data[i][xAxis] &&  d[color] == data[i][color];
+                    }
+                },
+                yAxis,
+                function(d) {
+                    isValueMatched = true;
+                    return data[i][yAxis];
+                });
+
+            if(isValueMatched){
+                var isIndexRemoved = false;
+
+                var index = allowedRemovableDataSet.indexOf(data[i][xAxis]);
+                if (index > -1) {
+                    // updated value matched in allowed removable values
+                    isIndexRemoved = true;
+                    allowedRemovableDataSet.splice(index, 1);
+                }
+
+                if(!isIndexRemoved){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD